students/Pirouz_N/lesson03/mailroom.py

```diff
@@ -0,0 +1,123 @@
+    print('\n')
+
+
+def print_list_of_donors():
+    """"This function will print list of donors."""
+    print('{:<30.30s}\t'.format('Donor Name'))
+    print('---------------------------------')
+
+    # printing donor names
+    for donor in donor_db:
+        # printing donor name
+        print('{:<30.30s}\t'.format(donor[0]))
+    print('\n')
+
+
+def write_thank_you(donerindex):
+    """"This function will write a thank to a donor.
+
+    Args:
+        donerindex: Is the index of the donor thank you card will be written to.
+    """
+    while True:
+
+        amount = input('Please enter your donation amount:')
+
+        try:
+            amount_float = float(amount)
+
+        except TypeError:
+            print('Entered amount cannot be accepted. Please enter a number.')
+        except ValueError:
+            print('Entered value cannot be accepted. Please enter a number.')
+        except:
+            print('Entered value cannot be accepted. Please try again.')
+        else:
+            donor_db[donerindex][1].append(amount_float)
+            print('Dear {},\n'.format(donor_db[donerindex][0]))
+            print('Thank you for your latest donation of ${:0.2f}. With this donation your overall donation has reached'
+                  ' ${:0.2f}. We are very grateful for all your {} donations, and we appreciate all your support. '
+                  'Please do not forget us in future we need because there is still more work that needs to be done and'
+                  ' we need your help to accomplish them.\n'.format(amount_float,
+                                                                    float(sum(donor_db[donerindex][1])),
+                                                                    len(donor_db[donerindex][1])))
+            print('Best Regards')
+            print('Pirouz Naghavi')
+            break
+
+
+def send_thank_you():
+    """"This function will type a thank card to the selected donor."""
+    while True:
+        option = input('Please type in the name of the donor you wish to thank.\n'
+                       'If you wish to see the list of all the donors, please type list.\n'
+                       'If you wish to add a new donor to the list, please type in their name.\n'
+                       'If you wish to return to the main menu please type exit and hit enter.\n'
+                       'Please type in your desired option and hit enter:\n')
+
+        if option == 'list':
+            print_list_of_donors()
+
+        elif option == 'exit':
+            break
+
+        else:
+            donor_names_list = [item[0] for item in donor_db]
+            if option in donor_names_list:
+                write_thank_you(donor_names_list.index(option))
+            else:
+                print('The new donor name you entered will be added to the donor table.')
+                donor_db.append((option, []))
+                write_thank_you(len(donor_names_list))
+            break
```

Comment:
please check assignment requirements, you should be asking for a donation amount and either updating existing donor with donation or adding a new donation for the donor.

Reply:
I am sorry if the name doesn't make this clear but write_thank_you does that. I will change the name to ask_donation_write_thank_you. I sometimes don't name functions or variables properly so if you notice that anywhere in current and future code submissions please let me know. I really appreciate any pointer you could give me that would make code look more like a more professional code. As you mentioned already python is not supposed to be verbose. I'll keep that in mind thank you.

Best Regards
Pirouz Naghavi